clang/docs/checkedc/Update-to-latest-LLVM-sources.md

```diff
@@ -1,81 +1,212 @@
+    git pull upstream <branch-point-of-12-on-main>
+```
+3. Execute  LLVM/Clang tests on the branch `updated_baseline` on your local
+machine to make sure that the upgrade is stable.
+
+4. Merge the `updated_baseline` branch back to the `baseline` branch.
+```
     git checkout baseline
-    git checkout -b updated-baseline
-    git pull mirror master
-
-## Run testing on your branched baseline branches.
-
-You can run testing locally or push your branched baseline branches to GitHub
-and use automated testing.  This will show you whether there are any unexpected
-failures in your baseline branch.
-
-If you use automated testing, make sure to clean the build directory first.
-Enough changes will likely have accumulated that things may go wrong without doing
-that first.
-
-## Branch your new baseline branches and merge master changes
-
-You can now branch your baseline branches to create a new master branch:
-
-    git checkout -b updated-master
+    git merge updated_baseline
+```
+5. Create a new branch, say, `updated_baseline_master_12` from the `baseline`
+branch.
+```
+    git checkout -b updated_baseline_master_12 baseline
+```
+6. Merge the `master` branch of the checkedc-clang repository into the
+`updated_baseline_master_12` branch. This merge may cause several merge
+conflicts and test case failures. After the merge command, git may suggest to
+you to set merge.renamelimit to a specific value. It is a good idea to set it as
+suggested and re-run the merge. The last section in this document gives
+guidelines on fixing the merge conflicts and test failures.
+```
+    git checkout updated_baseline_master_12
+    git merge master > merge_conflict_list.txt
+``` 
+7. The resolution of test failures may require you to cherry-pick commits from
+the upstream LLVM/Clang repository. Record the cherry-picked commits in the
+document LLVM-Upgrade-Notes.md. Let &lt;commit-to-cherry-pick&gt; be one
+such commit hash on branch &lt;cherry-pick-branch&gt; of the LLVM/Clang
+repository. 
+```
+    git fetch upstream <cherry-pick-branch>
+    git cherry-pick <commit-to-cherry-pick>
+```
+8. While the merge conflicts and test failures are being fixed, it is possible
+that the `master` branch has had several commits. Merge the `master` branch into
+`updated_baseline_master_12` branch, and resolve the new conflicts and failures
+if necessary.
+```
+    git merge master >> merge_conflicts_list.txt
+```
+9. After all the merge conflicts and test failures on the local machine are
+fixed, push the `baseline` and `updated_baseline_master_12` branches to the
+remote repository and excute the ADO tests on `updated_baseline_master_12`.
+```
+    git push origin baseline
+    git push origin updated_baseline_master_12
+```
+10. After the ADO tests are successful, merge the `updated_baseline_master_12`
+branch into `master` and push to the remote repository.
+```
+    git checkout master
+    git merge updated_baseline_master_12
```

Comment:
```suggestion
    git merge --ff-only updated_baseline_master_12
```
You probably also want to make sure that this command advances `master` exactly to `updated_baseline_master_12` (which was tested in the previous step) rather than generating a merge commit (which could fail the tests if Checked C feature development on `master` interfered somehow with the baseline upgrade).